src/webportal/package.json

```diff
@@ -84,19 +85,21 @@
     "slick-carousel": "~1.8.1",
     "strip-json-comments": "^2.0.1",
     "style-loader": "~0.19.0",
-    "uglifyjs-webpack-plugin": "~1.1.4",
+    "terser-webpack-plugin": "^1.2.3",
     "util": "~0.10.3",
-    "webpack": "~3.10.0",
+    "webpack": "~4.29.6",
+    "webpack-cli": "^3.3.0",
+    "webpack-dev-server": "^3.2.1",
     "winston": "~2.4.0"
   },
   "scripts": {
     "lint": "eslint --ext .js --ext .jsx src",
     "test": "npm run lint",
     "clean": "rimraf dist",
     "prebuild": "npm run clean && node config/preprocess.js",
-    "build": "webpack --config webpack.config.js --progress --profile",
-    "build:dev": "npm run build -- --debug --devtool eval --output-pathinfo --watch",
-    "prestart": "envsub --env-file .env --system src/app/env.js.template dist/scripts/env.js",
+    "build": "webpack --config webpack.config.js --progress --profile -p",
```

Comment:
If `webpack.config.js` is the default configuration path we could omit it like `dev` script.

The difference between `webpack --config` in `build` and `webpack-dev-server` in `dev` may lead confusing.

Reply:
Agree, maybe we could provide a ```webpack.dev.config.js``` for ```yarn dev``` and a ```webpack.prod.config.js``` for ```yarn build```.

Reply:
> 
> 
> Agree, maybe we could provide a `webpack.dev.config.js` for `yarn dev` and a `webpack.prod.config.js` for `yarn build` with different optimizations.

https://webpack.js.org/api/cli/#shortcuts
I think webpack's "-d" and "-p" flag is enough. The shortcuts includes source map and minify options.

Reply:
fixed.
use webpack-cli's "--mode" option.